Source/Selling/Mixins/Item.lua

```diff
@@ -0,0 +1,153 @@
+AuctionatorItemSellingMixin = {}
+
+local AUCTIONATOR_ITEM_EVENTS = {
+  "ITEM_SEARCH_RESULTS_UPDATED",
+  "ITEM_SEARCH_RESULTS_ADDED"
+}
+
+function AuctionatorItemSellingMixin:Initialize()
+  Auctionator.Debug.Message("AuctionatorItemSellingMixin:Initialize()")
+
+  hooksecurefunc(
+    AuctionHouseFrame.ItemSellFrame.ItemDisplay,
+    "SetItemInternal",
+    function()
+      self:SetDuration(
+        AuctionHouseFrame.ItemSellFrame.DurationDropDown,
+        tonumber(Auctionator.Config.Get(Auctionator.Config.Options.ITEM_AUCTION_DURATION))
+      )
+
+      FrameUtil.RegisterFrameForEvents(self, AUCTIONATOR_ITEM_EVENTS)
+    end
+  )
+end
+
+function AuctionatorItemSellingMixin:AggregateItemResults(itemKey, itemCount, itemLevel)
+  local results = {}
+  local currentResult
+
+  for index = 1, itemCount do
+    currentResult = C_AuctionHouse.GetItemSearchResultInfo(itemKey, index)
+
+    if currentResult == nil then
+      break
+    elseif currentResult.itemKey.itemLevel == itemLevel then
+      -- Only aggregate items at the same iLvl as the posted piece
+      table.insert(results, currentResult)
+    end
+  end
+
+  return results
+end
+
+local function copyKey(originalItemKey)
+  return {
+    itemLevel = originalItemKey.itemLevel,
+    itemSuffix = originalItemKey.itemSuffix,
+    itemID = originalItemKey.itemID,
+    battlePetSpeciesID = originalItemKey.battlePetSpeciesID
+  }
+end
```

Comment:
That's one way to do it, but a generic copy table function would work too. Not necessary, just something to thing about
```lua
function CloneTable(original)
  result = {}
  for key, value pairs(original) do
    result[key] = value
  end
end
```